### DISCUSSION TITLE
Create usd stage from file using python

### DISCUSSION THREAD

**Discussion by raphaeljnwave** [Autodesk/maya-usd#1847] [Q&A] [answered]
Hi,

I am trying to load usd file in Maya using python (trying to reproduce Create -> Universal Scene Description (USD) -> Stage From File ...) but I only succeeded to convert an usd file into maya object, with no usd stage. By looking the doc, it appears that the -uac argument from mayaUSDImport is what I am looking for, but it does not do the job : cmds.mayaUSDImport( file=path, useAsAnimationCache=True)
Can you please highlight me on this problem ?

Thank you,
Raphael

**Comment by brycegbrazen** [comment, marked as answer]
@raphaeljnwave Have you tried doing this?

``` 
shape_node = cmds.createNode('mayaUsdProxyShape')
cmds.setAttr('{}.filePath'.format(shape_node), filename, type='string')
```

**Reply by raphaeljnwave** [reply]
Thanks ! Indeed, I did not thought about creating directly the node :)

**Reply by ppt-adsk** [reply]
Have a look at what
https://github.com/Autodesk/maya-usd/blob/c114b4b2a803c993a0dadaf20d836ca97fc32e8b/plugin/adsk/scripts/mayaUsd_createStageFromFile.mel#L118
does.  We go a bit further and populate some attributes, as well as connect to the time node.  It's admittedly awkward that this is only available in MEL, but you could create your own Python function that calls this MEL procedure.  Hope this helps.

**Reply by raphaeljnwave** [reply]
Yes it will help  for sure ! Thanks @ppt-adsk
